examples/10.dynamic_configuration/config_broker.h

```diff
@@ -0,0 +1,61 @@
+// Copyright Microsoft and CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+
+// Contributed by Configured Things Ltd
+
+#include "cdefs.h"
+#include "compartment-macros.h"
+#include "token.h"
+#include <compartment.h>
+
+typedef char ConfigId[16];
+#define MAX_CONFIG_IDS 8
+
+struct ConfigToken
+{
+	bool     is_source;
+	uint16_t id;
+	uint8_t  count;
+	ConfigId configId[MAX_CONFIG_IDS];
```

Comment:
This can be a variable length thing.  We do that [in the network stack](https://github.com/CHERIoT-Platform/network-stack/blob/eeeecc2946a6c591ca259d36f8c0df53fe09440f/include/NetAPI.h#L250).

This is the [accompanying rego policy](https://github.com/CHERIoT-Platform/network-stack/blob/eeeecc2946a6c591ca259d36f8c0df53fe09440f/network_stack.rego#L22) that checks that these are valid (including that the length matches the number of items).

Reply:
Ah, yes, defining a structure that contains an array of variable-length strings is tricky.  I'd be tempted to have a separate capability for each, so that you don't need to separately state the thing you want to publish / subscribe *and* the name, and so that you preserve the principle of intentional use.

Reply:
I've posted some updates - still WIP so haven't acted on all of the suggestions, but the main change is to move to separate capabilities for each config item